tests/Sharpliner.Tests/AzureDevOps/DependencyVariableReferenceTests.cs

```diff
@@ -0,0 +1,390 @@
+    /// </summary>
+    private class StageToStageDeployResourceDependencyVariableReferenceTest_Template : StageTemplateDefinition
+    {
+        public override string TargetFile => "stages.yml";
+
+        private string setterStageName = "build";
+        private string setterJobName = "build_job";
+        private string setterStepName = "setRunTests";
+        private string setterResourceName = "winVM2";
+        private string sharedVariableName = "runTests";
+
+        private string getterStageName = "test";
+        private string getterJobName = "A";
+
+        public override ConditionedList<Stage> Definition =>
+        [
+            new Stage(setterStageName)
+            {
+                Jobs =
+                {
+                    new DeploymentJob(setterJobName)
+                    {
+                        Environment = new Sharpliner.AzureDevOps.Environment("vmtest", setterResourceName)
+                        {
+                            ResourceType = ResourceType.VirtualMachine
+                        },
+                        Strategy = new RunOnceStrategy
+                        {
+                            Deploy = new()
+                            {
+                                Steps =
+                                [
+                                    new PwshTaskBuilder().Inline(
+                                        $"""
+                                        ${sharedVariableName} = "true"
+                                        echo "setting {sharedVariableName}: ${sharedVariableName}"
+                                        echo "##vso[task.setvariable variable={sharedVariableName};isOutput=true]${sharedVariableName}"
+                                        """
+                                    ) with
+                                    {
+                                        Name = setterStepName
+                                    }
+                                ]
+                            }
+                        }
+                    },
+                }
+            },
+            new Stage(getterStageName) {
+                Condition = Equal(dependencies.stage.deploy[setterStageName, setterJobName, setterStepName, sharedVariableName, setterResourceName], "true"),
+                DependsOn = setterStageName,
+                Jobs =
+                {
+                    new Job(getterJobName)
+                    {
+                        Steps =
+                        {
+                            Script.Inline($"echo hello from Job {getterJobName}")
+                        }
+                    },
+                }
+            }
+        ];
+    }
+
+    // This test should create a variable in one job and reference it in the condition for another stage.
```

Comment:
consider removing the comments on these methods, they don't provide additional information to the test names :)